test/replicatedmap/ReplicatedMapProxyTest.js

```diff
@@ -29,245 +29,168 @@ describe('ReplicatedMapProxyTest', function () {
 
-    before(function () {
+    before(async function () {
         this.timeout(10000);
         const config = fs.readFileSync(path.join(__dirname, 'hazelcast_replicatedmap.xml'), 'utf8');
-        return RC.createCluster(null, config).then(function (response) {
-            cluster = response;
-            return RC.startMember(cluster.id);
-        }).then(function () {
-            return Client.newHazelcastClient({ clusterName: cluster.id });
-        }).then(function (hazelcastClient) {
-            client = hazelcastClient;
-        });
+        cluster = await RC.createCluster(null, config);
+        await RC.startMember(cluster.id);
+        client = await Client.newHazelcastClient({ clusterName: cluster.id });
     });
 
-    beforeEach(function () {
-        return client.getReplicatedMap('test').then(function (mp) {
-            rm = mp;
-        });
+    beforeEach(async function () {
+        rm = await client.getReplicatedMap('test');
     });
 
-    afterEach(function () {
+    afterEach(async function () {
         return rm.destroy();
     });
 
-    after(function () {
-        return client.shutdown()
-            .then(() => RC.terminateCluster(cluster.id));
+    after(async function () {
+        await client.shutdown();
+        return RC.terminateCluster(cluster.id);
     });
 
-    it('puts one entry and gets one entry', function () {
-        return rm.put('key', 'value', ONE_HOUR)
-            .then(function () {
-                return rm.get('key')
-            })
-            .then(function (val) {
-                expect(val).to.equal('value');
-            });
+    it('puts one entry and gets one entry', async function () {
+        await rm.put('key', 'value', ONE_HOUR);
+        const val = await rm.get('key');
+        expect(val).to.equal('value');
     });
 
-    it('puts entry with 0 ttl', function () {
-        return rm.put('zerottl', 'value')
-            .then(function () {
-                return rm.get('zerottl');
-            })
-            .then(function (val) {
-                expect(val).to.equal('value');
-            });
+    it('puts entry with 0 ttl', async function () {
+        await rm.put('zerottl', 'value');
+        const val = await rm.get('zerottl');
+        expect(val).to.equal('value');
     });
 
-    it('puts entry with 1000 ttl', function () {
-        return rm.put('1000ttl', 'value', 1000)
-            .then(function () {
-                return new Promise(function (resolve) {
-                    setTimeout(function () {
-                        resolve();
-                    }, 2000)
-                });
-            })
-            .then(function () {
-                return rm.get('1000ttl');
-            })
-            .then(function (val) {
-                expect(val).to.be.null;
-            });
+    it('puts entry with 1000 ttl', async function () {
+        await rm.put('1000ttl', 'value', 1000);
+        await new Promise(function (resolve) {
```

Comment:
Minor: I know the test is like that before but we can use `promiseWaitMilliseconds` from the `test/Util.js`

Reply:
Addressed.